k8s/sonarqube/README.md

```diff
@@ -0,0 +1,439 @@
+[on-screen instructions](https://console.cloud.google.com/marketplace/details/google/sonarqube).
+
+## Command line instructions
+
+You can use [Google Cloud Shell](https://cloud.google.com/shell/) or a local workstation in the
+further instructions.
+
+[![Open in Cloud Shell](http://gstatic.com/cloudssh/images/open-btn.svg)](https://console.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/click-to-deploy&cloudshell_working_dir=k8s/sample-app)
+
+### Prerequisites
+
+#### Set up command line tools
+
+You'll need the following tools in your environment:
+
+- [gcloud](https://cloud.google.com/sdk/gcloud/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [docker](https://docs.docker.com/install/)
+- [openssl](https://www.openssl.org/)
+- [helm](https://helm.sh/docs/using_helm/#installing-helm)
+
+Configure `gcloud` as a Docker credential helper:
+```shell
+gcloud auth configure-docker
+```
+#### Create a Google Kubernetes Engine cluster
+
+Create a cluster from the command line. If you already have a cluster that
+you want to use, this step is optional.
+
+```shell
+export CLUSTER=app-cluster
+export ZONE=us-west1-a
+
+gcloud container clusters create "$CLUSTER" --zone "$ZONE"
+```
+
+#### Configure kubectl to connect to the cluster
+
+```shell
+gcloud container clusters get-credentials "$CLUSTER" --zone "$ZONE"
+```
+
+#### Clone this repo
+
+Clone this repo and the associated tools repo:
+
+```shell
+git clone --recursive https://github.com/GoogleCloudPlatform/click-to-deploy.git
+```
+
+#### Install the Application resource definition
+
+An Application resource is a collection of individual Kubernetes components,
+such as Services, Deployments, and so on, that you can manage as a group.
+
+To set up your cluster to understand Application resources, run the following command:
+
+```shell
+kubectl apply -f "https://raw.githubusercontent.com/GoogleCloudPlatform/marketplace-k8s-app-tools/master/crd/app-crd.yaml"
+```
+
+You need to run this command once for each cluster.
+
+The Application resource is defined by the
+[Kubernetes SIG-apps](https://github.com/kubernetes/community/tree/master/sig-apps) community. The source code can be found on
+[github.com/kubernetes-sigs/application](https://github.com/kubernetes-sigs/application).
+
+### Install the application
+
+Navigate to the `sonarqube` directory:
+
+```shell
+cd click-to-deploy/k8s/sonarqube
+```
+
+#### Configure the application with environment variables
+
+Choose an instance name and
+[namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/)
+for the application. In most cases, you can use the `default` namespace.
+
+```shell
+export APP_INSTANCE_NAME=sonarqube-1
+export NAMESPACE=default
+```
+
+
+Configure the container image:
+
+```shell
+TAG=9.6
```

Comment:
It really doesn't make sense the `TAG` var is here. You don't use it.

Please point to the images below to the images used by k8s app (the currents ones are before republishing).

For example (copied from WP):

```
TAG=latest
export IMAGE_WORDPRESS="marketplace.gcr.io/google/wordpress:${TAG}"
export IMAGE_APACHE_EXPORTER="marketplace.gcr.io/google/wordpress/apache-exporter:${TAG}"
export IMAGE_MYSQL="marketplace.gcr.io/google/wordpress/mysql:${TAG}"
export IMAGE_MYSQL_EXPORTER="marketplace.gcr.io/google/wordpress/mysqld-exporter:${TAG}"
export IMAGE_METRICS_EXPORTER="marketplace.gcr.io/google/wordpress/prometheus-to-sd:${TAG}"
```